tests/network/hotplug.go

```diff
@@ -405,6 +405,122 @@ var _ = SIGDescribe("nic-hotunplug", func() {
 			Entry("Migration based", decorators.MigrationBasedHotplugNICs, migrationBased),
 		)
 	})
+
+	Context("a running VM", func() {
+		var vm *v1.VirtualMachine
+		var vmi *v1.VirtualMachineInstance
+
+		const guestNICName = "eth2"
+
+		BeforeEach(func() {
+			By("creating a NAD")
+			Expect(createBridgeNetworkAttachmentDefinition(
+				testsuite.GetTestNamespace(nil), networkName, linuxBridgeName)).To(Succeed())
+
+			By("running a VM")
+			opts := append(
+				libvmi.WithMasqueradeNetworking(),
+				libvmi.WithNetwork(libvmi.MultusNetwork(linuxBridgeNetworkName1, networkName)),
+				libvmi.WithNetwork(libvmi.MultusNetwork(linuxBridgeNetworkName2, networkName)),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding(linuxBridgeNetworkName1)),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding(linuxBridgeNetworkName2)),
+			)
+			vmi = libvmi.NewAlpineWithTestTooling(opts...)
+			vm = tests.NewRandomVirtualMachine(vmi, true)
+
+			var err error
+			vm, err = kubevirt.Client().VirtualMachine(testsuite.GetTestNamespace(nil)).Create(context.Background(), vm)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(func() error {
+				vmi, err = kubevirt.Client().VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &metav1.GetOptions{})
+				return err
+			}, 120*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+
+			libwait.WaitForSuccessfulVMIStart(vmi)
+
+			By("unplug interface")
+			Expect(kubevirt.Client().VirtualMachine(vm.Namespace).RemoveInterface(context.Background(), vm.Name,
+				&v1.RemoveInterfaceOptions{Name: linuxBridgeNetworkName2},
+			)).To(Succeed())
+		})
+
+		DescribeTable("hot-unplug network interface succeed", func(plugMethod hotplugMethod) {
+			By("wait for vmi spec interface have absent state")
+			Eventually(func() v1.InterfaceState {
+				var err error
+				vmi, err = kubevirt.Client().VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.Name, &metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				iface := vmispec.LookupInterfaceByName(vmi.Spec.Domain.Devices.Interfaces, linuxBridgeNetworkName2)
+				return iface.State
+			}, 30*time.Second).Should(Equal(v1.InterfaceStateAbsent))
+
+			By("verify unplugged interface is not reported in the VMI status")
+			vmi = verifyDynamicInterfaceChange(vmi, plugMethod)
+
+			By("restarting the VM")
+			Expect(kubevirt.Client().VirtualMachine(vm.Namespace).Restart(context.Background(), vm.Name, &v1.RestartOptions{})).To(Succeed())
+
+			By("wait for new VMI to start")
+			var newVMI *v1.VirtualMachineInstance
+			var err error
+			Eventually(func() v1.VirtualMachineInstancePhase {
+				newVMI, err = kubevirt.Client().VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &metav1.GetOptions{})
+				if err != nil || vmi.UID == newVMI.UID {
+					return v1.VmPhaseUnset
+				}
+				return newVMI.Status.Phase
+			}, 90*time.Second, 1*time.Second).Should(Equal(v1.Running))
```

Comment:
I do not understand this waiting step... It was not used before, so why now?

Reply:
The phase part confuses me. Originally we just waited for the VMI object to exist, not to wait for a specific phase. If it is already used elsewhere here, fine.

Reply:
Phase checking is removed.